ui/src/app/store/device/selectors.ts

```diff
@@ -1,14 +1,61 @@
+import { createSelector } from "reselect";
+
+import type { RootState } from "../root/types";
+
 import { DeviceMeta } from "app/store/device/types";
-import type { Device, DeviceState } from "app/store/device/types";
+import type { Device, DeviceState, DeviceStatus } from "app/store/device/types";
 import { generateBaseSelectors } from "app/store/utils";
 
 const searchFunction = (device: Device, term: string) =>
   device.fqdn.includes(term);
 
-const selectors = generateBaseSelectors<DeviceState, Device, DeviceMeta.PK>(
-  DeviceMeta.MODEL,
-  DeviceMeta.PK,
-  searchFunction
+const defaultSelectors = generateBaseSelectors<
+  DeviceState,
+  Device,
+  DeviceMeta.PK
+>(DeviceMeta.MODEL, DeviceMeta.PK, searchFunction);
+
+/**
+ * Get the device state object.
+ * @param state - The redux state.
+ * @returns The device state.
+ */
+const deviceState = (state: RootState): DeviceState => state[DeviceMeta.MODEL];
+
+/**
+ * Get the devices statuses.
+ * @param state - The redux state.
+ * @returns The device statuses.
+ */
+const statuses = createSelector(
+  [deviceState],
+  (deviceState) => deviceState.statuses
 );
 
+/**
+ * Get the statuses for a device.
+ * @param state - The redux state.
+ * @param id - A device's system id.
+ * @returns The device's statuses
+ */
+const getStatusForDevice = createSelector(
+  [
+    statuses,
+    (
+      _state: RootState,
+      id: Device[DeviceMeta.PK] | null,
+      status: keyof DeviceStatus
+    ) => ({
+      id,
+      status,
+    }),
+  ],
+  (allStatuses, { id, status }) => (id ? allStatuses[id][status] : false)
```

Comment:
Should we be checking `allStatuses[id]` exists before trying to access `status`?

Reply:
Yep, done. Not sure why typescript didn't complain.